### ISSUE TITLE
What is the alternative to "--index-filter" and "--tag-name-filter"?

### ISSUE THREAD

**Issue by maravento** [newren/git-filter-repo#97]
What is the alternative to "--index-filter" and "--tag-name-filter"?

**Note**: i replace "git filter-branch" with "git filter-repo"

`#!/bin/bash`
`project="myrepo"`
`file="bigfile.tar.gz"`
`cd $project`
`git filter-repo --force --index-filter 'git rm -r --cached --ignore-unmatch '"$file"'' --prune-empty auto --tag-name-filter cat -- --all`

out:

`git-filter-repo: error: unrecognized arguments: --index-filter git rm -r --cached --ignore-unmatch bigfile.tar.gz --tag-name-filter cat -- --all`

`git filter-repo --version`
`ffc6fa9265d8`
`git --version`
`git version 2.25.1`

thanks

**Comment by newren** [comment]
There's a [cheat sheet for converting filter-branch commands to filter-repo ones](https://github.com/newren/git-filter-repo/blob/master/Documentation/converting-from-filter-branch.md#cheat-sheet-conversion-of-examples-from-the-filter-branch-manpage) that should help you with these questions in general, however the following flags from filter-branch roughly translate as follows:

* --prune-empty: If you just want to prune commits which become empty then you don't even need to specify this flag.  If you want to prune commits that started empty in your repo, then you need to specify `--prune-empty always`.
* --tag-name-filter: If you are just specifying `--tag-name-filter cat`, then the correct translation is to specify no extra flags.  The fact that filter-branch required that was evidence that it was retarded; it should have been handled automatically.  (If you use something other than 'cat', i.e. you really are renaming tags, then there's a `--tag-rename` option.)
* `-- --all`: This was another evidence that filter-branch was retarded in making users specify things that should have just been the default.  Just drop it.
* `--index-filter ...`.  There's not a direct equivalent, but people almost always use this just to prune or keep files based on filename, in which case you want to use the various --path* options and, if specifying paths to prune rather than paths to keep, use --invert-paths.

Anyway, in summary, the equivalent of
```
git filter-branch --index-filter 'git rm -r --cached --ignore-unmatch bigfile.tar.gz' --prune-empty --tag-name-filter cat -- --all
```

is just
```
git filter-repo --path bigfile.tar.gz --invert-paths
```

**Comment by maravento** [comment]
thanks
